getting-started/basicapp/README.md

```diff
@@ -0,0 +1,362 @@
+[Stackdriver Profiler](https://cloud.google.com/profiler/) and 
+[Stackdriver Trace](https://cloud.google.com/trace/).
+
+
+## Project Setup
+The prerequisites for running the example locally are 
+[Go](https://golang.org/dl/), the 
+[gcloud](https://cloud.google.com/sdk/gcloud/) command line tool, and Docker
+community edition. If you prefer not to install gcloud and Docker locally you
+can use the [Cloud Shell](https://cloud.google.com/shell/docs/) instead. 
+
+To run this example, you will need to create a Google Cloud project with billing
+enabled. You will also need to enable the Cloud SQL API, Cloud SQL Admin API,
+the Stackdriver Profiling API, and the Stackdriver Trace API.
+
+If you are trying this out in your own workspace you can get the code with the
+git clone command
+```
+git clone https://github.com/GoogleCloudPlatform/golang-samples.git
+cd getting-started/basicapp
+```
+
+If you want to try changing code then you may want to save your code in a
+private [Google Cloud Source
+repository](https://cloud.google.com/source-repositories/docs/) with the
+commands
+```
+gcloud source repos create basicapp
+git add .
+git commit -m "My private copy"
+```
+
+## Local development, mocks, and unit testing
+You can run locally and unit test using a mock of the messaging service.
+
+### Running locally
+To run the app locally commands export an environment variable for use of mocks
+instead of a real database:
+```
+export MESSAGE_SERVICE=mock
+go get -d -v ./...
+go run basicwebapp.go
+```
+
+Check that the application successfully responds to a HTTP request:
+```
+curl -I http://localhost:8080
+```
+
+You can now use curl to send a message to your friend a message like
+```
+curl "http://localhost:8080/send?user=Friend1&friend=Friend2&text=We+miss+you!"
+```
+
+Your friend can also check their messages with a URL like:
+```
+curl "http://localhost:8080/messages?user=Friend2"
+```
+
+With a mock service we lose the messages as soon as the app is stopped. Unset
+the environment variable for use of mocks with the command
+```
+unset MESSAGE_SERVICE
+```
+
+## Unit tests
+To run unit tests use the two commands, for each of the main and services
+packages:
+```
+go test
+go test -v github.com/GoogleCloudPlatform/golang-samples/getting-started/basicapp/services
+```
+
+Check that the tests pass. The Go
+[httptest](https://golang.org/pkg/net/http/httptest/) package is used to
+simulate HTTP requests and writers in some of the unit tests.
+
+Run the build and execute unit tests using Cloud Build
+```
+gcloud builds submit --config build/cb-unittest.yaml .
+```
+
+View the build and test results in the cloud console under the Cloud Build menu.
+
+## Database
+### Setup
+This section is based on the instructions at [Using Cloud SQL with
+Go](https://cloud.google.com/go/getting-started/using-cloud-sql). Use the
+commands below from the Cloud Shell. Make sure that the Cloud SQL API and the
+Cloud SQL Admin API are enabled.
```

Comment:
Links to enable APIs?

Reply:
Consolidated links to enable APIs at the top